shouldupdate.js

```diff
@@ -67,36 +67,22 @@ function factory (methods) {
   return shouldComponentUpdate;
 
   function shouldComponentUpdate (nextProps, nextState) {
-    var isNotIgnorable = not(or(isStatics, isChildren));
-
-    var nextCursors    = filter(nextProps, isNotIgnorable),
-        currentCursors = filter(this.props, isNotIgnorable);
-
-    var nextCursorsKeys    = Object.keys(nextCursors),
-        currentCursorsKeys = Object.keys(currentCursors);
-
-    if (currentCursorsKeys.length !== nextCursorsKeys.length) {
-      if (debug) debug.call(this, 'shouldComponentUpdate => true (number of cursors differ)');
-      return true;
+    if (nextProps === this.props) {
+      if (debug) debug.call(this, 'shouldComponentUpdate => false (equal input)');
+      return false;
     }
 
-    if (hasDifferentKeys(currentCursorsKeys, currentCursors, nextCursors)) {
-      if (debug) debug.call(this, 'shouldComponentUpdate => true (cursors have different keys)');
-      return true;
-    }
-
-    if (hasChangedCursors(currentCursors, nextCursors)) {
-      if (debug) debug.call(this, 'shouldComponentUpdate => true (cursors have changed)');
-      return true;
-    }
+    var isNotIgnorable = not(or(isStatics, isChildren));
+    var nextProps    = filter(nextProps, isNotIgnorable),
+        currentProps = filter(this.props, isNotIgnorable);
```

Comment:
Move this below `_isEqualState` block for better placement:

```js
    var nextProps    = filter(nextProps, isNotIgnorable),
        currentProps = filter(this.props, isNotIgnorable);
```

Reasoning is that it could be a pointless computation if `!_isEqualState(this.state, nextState) === true`.